### ISSUE TITLE
nimble file should probably not use @#head

### ISSUE THREAD

**Issue by c-blake** [jangko/nimAGG#59]
See https://github.com/c-blake/hldiff/issues/1#issuecomment-682113924 and subsequent.

Seems you wanted some May 2017 version of sdl2_nim.  The git hash you probably want is something like 213ee80f8fc4ec2d4dfedd02ee091dbfd336983f

**Comment by jangko** [comment]
breaking changes in sdl2_nim?

**Comment by c-blake** [comment]
It's really more an issue that nimble never updates the `@#head` version after it is first installed.  This surprised me. I would have thought it was "always updated".  Instead they are telling us in the first paragraph of that linked comment by @genotrance that `@#head` is actually "never updated" after the initial install.

In case the perils of "never updated" are not obvious..As a concrete example, if someone installed your package on May 20, 2017 (the day after you added `sdl2_nim@#head` to your nimble) and never removed that `@#head` version from `pkgs/` then it will never have been updated since 2017.  But it's even worse!  If *another package* did `requires sdl2_nim@#head` *before you*, and a user had it installed in the shared `pkgs/` then it may have caused that `@#head` slot to be allocated and never updated ever after.  If `nimAGG` needed `sdl2_nim` features/bug fixes from any time after this unknowable-to-you point in time it would fail.

In short, because of the way `nimble` operates and `@#head`, in some `nimAGG` user's shared install `pkgs/` directory the installed and used `sdl2_nim@#head` could be as old as 2013 or as new as 2020.
